clarifai/client/input.py

```diff
@@ -729,6 +731,37 @@ def delete_inputs(self, inputs: List[Input]) -> None:
       raise Exception(response.status)
     self.logger.info("\nInputs Deleted\n%s", response.status)
 
+  def download_inputs(self, inputs: List[Input]) -> List[bytes]:
+    """Download list of input objects from the app.
+
+    Args:
+        input_ids (Input): List of input objects to download.
+
+    Example:
+        >>> from clarifai.client.user import User
+        >>> input_obj = User(user_id="user_id").app(app_id="app_id").inputs()
+        >>> input_obj.download_inputs(list(input_obj.list_inputs()))
+    """
+    if not isinstance(inputs, list):
+      raise UserError("input_ids must be a list of input ids")
+    final_inputs = []
+    #initiate session
+    session = requests.Session()
+    retries = Retry(total=3, backoff_factor=0.1, status_forcelist=[500, 502, 503, 504])
+    session.mount('https://', HTTPAdapter(max_retries=retries))
+    session.headers.update({'Authorization': self.metadata[0][1]})
+    # download inputs
+    data_types = ['image', 'video', 'audio', 'text']
+    for input in inputs:
+      for data_type in data_types:
+        url = getattr(input.data, data_type).url
+        if url:
+          response = session.get(url, stream=True)
```

Comment:
As this is io intensive, we can make use of `ThreadPoolExecutor`

Reply:
For bulk download, we have dataset export function, there we use Threadpoolexecutor. This is majorly for minimal input download for quick viewing/storing. I think its not needed here. WDYT?